# Importa as libs necessarias

In [3]:
import pandas as pd
import json

# Ler o arquivo json

In [4]:
df = pd.read_json('www.properly.ca.har')

Volta o valor dp df que foi definido acima

In [5]:
df

log
creator      {'name': 'WebInspector', 'version': '537.36'}
entries  [{'_initiator': {'type': 'other'}, '_priority'...
pages    [{'startedDateTime': '2023-01-05T17:11:11.167Z...
version                                                1.2

In [6]:
df[df['pages']]

KeyError: 'pages'

Adicionar uma primeira coluna de indexacao, que comeca em zero

In [7]:
df = df.reset_index()

parece estar localizando a coluna 'log' e o index 'pages' e retornando o valor 

In [9]:
df[df['log']['index']=='pages']

KeyError: 'index'

selecionar a linha/index 'entries' e coluna 'log' e exportar pra um arquivo json chamado tst

In [10]:
df[df['index']=='entries']['log'].to_json('tst.json')

parece estar selecionando a mesma linha/coluna usada acima, mais nada?

In [ ]:
df[df['index']=='entries']['log']

cria um dataframe 'df2', que vai assumior o valor do aqruivo json criado acima

In [11]:
df2 = pd.read_json('tst.json')

importar o modulo sys que serve pra obter informacoes e ja usa ele ara obter o tamanho em bytes do dataframe df2

In [16]:
import sys
sys.getsizeof(df2)

2102848

atribui o conteudo da segunda coluna / primeira linha do df2 para fl

In [12]:
fl = df2[1][0]

usando a mesma funca de sys para ver o tamanho em bytes de fl

In [ ]:
sys.getsizeof(fl)

fl?

In [18]:
fl

{'_initiator': {'type': 'other'},
 '_priority': 'VeryHigh',
 '_resourceType': 'document',
 'cache': {},
 'connection': '40345',
 'pageref': 'page_1',
 'request': {'method': 'GET',
  'url': 'https://www.properly.ca/buy/search/toronto/?bounds=%7B%22sw%22%3A%7B%22lat%22%3A43.57821514343695%2C%22lng%22%3A-79.71032656047284%7D%2C%22ne%22%3A%7B%22lat%22%3A43.792721126438295%2C%22lng%22%3A-79.05114687297284%7D%7D&sorting=prod-global-listings-close-date&showSold=true',
  'httpVersion': 'http/2.0',
  'headers': [{'name': ':authority', 'value': 'www.properly.ca'},
   {'name': ':method', 'value': 'GET'},
   {'name': ':path',
    'value': '/buy/search/toronto/?bounds=%7B%22sw%22%3A%7B%22lat%22%3A43.57821514343695%2C%22lng%22%3A-79.71032656047284%7D%2C%22ne%22%3A%7B%22lat%22%3A43.792721126438295%2C%22lng%22%3A-79.05114687297284%7D%7D&sorting=prod-global-listings-close-date&showSold=true'},
   {'name': ':scheme', 'value': 'https'},
   {'name': 'accept',
    'value': 'text/html,application/xhtml+xml,

selecionou o mesmo valor acima e salvou para um arquivo json 'first line'

In [13]:
df2[1][0].to_json('first_line.json')

AttributeError: 'dict' object has no attribute 'to_json'

retornar o valor de df2 so pra confirmar, ele tem apenas os valores da linha 'entries' 

In [14]:
df2

1
0     {'_initiator': {'type': 'other'}, '_priority':...
1     {'_fromCache': 'memory', '_initiator': {'type'...
2     {'_fromCache': 'memory', '_initiator': {'type'...
3     {'_fromCache': 'memory', '_initiator': {'type'...
4     {'_fromCache': 'memory', '_initiator': {'type'...
...                                                 ...
4607  {'_initiator': {'type': 'script', 'stack': {'c...
4608  {'_initiator': {'type': 'script', 'stack': {'c...
4609  {'_initiator': {'type': 'script', 'stack': {'c...
4610  {'_initiator': {'type': 'script', 'stack': {'c...
4611  {'_initiator': {'type': 'script', 'stack': {'c...

[4612 rows x 1 columns]

contar a quantidade de linhas do df2

In [23]:
df2.count()

1    4612
dtype: int64

retorna o valor das linhas comecando do topo, como nao esta especificado () voltaria as cinco primeiras

In [15]:
df2.head()

1
0  {'_initiator': {'type': 'other'}, '_priority':...
1  {'_fromCache': 'memory', '_initiator': {'type'...
2  {'_fromCache': 'memory', '_initiator': {'type'...
3  {'_fromCache': 'memory', '_initiator': {'type'...
4  {'_fromCache': 'memory', '_initiator': {'type'...

renomear a coluna que chamava '1' para 'text'

In [16]:
df2 = df2.rename(columns = {1:'text'})

cria 'js_ex' que vai ter o valor da coluna 'text' / primeira linha e
'js_ex2' que vai ter o valor da coluna 'text' / segunda linha




In [17]:
js_ex = df2['text'][0]
js_ex2 = df2['text'][1]

criar 'json_object', que vai fazer uma serie mais ordenada em formato json da zona que estao os valores de 'js_ex' usando a funcao json.dumps. Em seguida,salvou esse dump organizado 

In [18]:
# Serializing json
json_object = json.dumps(js_ex)
 
# Writing to sample.json
with open("sample.json", "w") as outfile:
    outfile.write(json_object)

idem

In [19]:
# Serializing json
json_object2 = json.dumps(js_ex2)
 
# Writing to sample.json
with open("sample2.json", "w") as outfile:
    outfile.write(json_object2)

pra cada item i no range de 0 ao comprimento de df2 (o arquivo todo):

-cria 'js' que tera o valor localizado na coluna 'text' / linha 'i'

-converte o valor baguncado em uma seria mais ordenada no formato json

-define 'item' como o string contido em cada i

-Exporta um outfile com nome sample_'cada i'.json, dando overwrite se necessario, aonde o outfile eh o conteudo do 'json-object' que foi criado com a funcao json.dumps acima

In [29]:
for i in range(0,len(df2)):
    js = df2['text'][i]
    json_object = json.dumps(js)
    item = str(i)
    with open (f'sample_{item}.json','w') as outfile:
        outfile.write(json_object)

No dataframe df2 coluna text, conta quantas vezes o conteudo de cada linha se repete, 3106+1469+35+2=4611 que eh o mesmo numero de linhas do dataframe

In [20]:
df2['text'].str.len().value_counts()

12    3106
11    1469
13      35
10       2
Name: text, dtype: int64

In [42]:
_initiator,
"_priority": "High",
"_resourceType": "xhr",
"cache": {},
"connection": "41142",
"pageref": "page_1",
request
response:{
    status,
    statusText,
    httpVersion,
    headers,
    cookies,
    content:{
        size,
        mimetype,
        compression,
        text,
        redirectURL,
        headersSize,
        bodySize,
        _transferSize,
        _error
    }
}

SyntaxError: illegal target for annotation (3406225311.py, line 2)

In [40]:
res = test_dict.get('Gfg', {}).get('is')

NameError: name 'test_dict' is not defined

coloquei df2 aqui pra visualizar

In [21]:
df2

text
0     {'_initiator': {'type': 'other'}, '_priority':...
1     {'_fromCache': 'memory', '_initiator': {'type'...
2     {'_fromCache': 'memory', '_initiator': {'type'...
3     {'_fromCache': 'memory', '_initiator': {'type'...
4     {'_fromCache': 'memory', '_initiator': {'type'...
...                                                 ...
4607  {'_initiator': {'type': 'script', 'stack': {'c...
4608  {'_initiator': {'type': 'script', 'stack': {'c...
4609  {'_initiator': {'type': 'script', 'stack': {'c...
4610  {'_initiator': {'type': 'script', 'stack': {'c...
4611  {'_initiator': {'type': 'script', 'stack': {'c...

[4612 rows x 1 columns]

In [22]:
df2['text'][3]

{'_fromCache': 'memory',
 '_initiator': {'type': 'parser',
  'url': 'https://www.properly.ca/buy/search/toronto/?bounds=%7B%22sw%22%3A%7B%22lat%22%3A43.57821514343695%2C%22lng%22%3A-79.71032656047284%7D%2C%22ne%22%3A%7B%22lat%22%3A43.792721126438295%2C%22lng%22%3A-79.05114687297284%7D%7D&sorting=prod-global-listings-close-date&showSold=true',
  'lineNumber': 156},
 '_priority': 'Low',
 '_resourceType': 'script',
 'cache': {},
 'connection': '40345',
 'pageref': 'page_1',
 'request': {'method': 'GET',
  'url': 'https://www.properly.ca/_next/static/chunks/main-0d663b80e19542b1.js',
  'httpVersion': 'http/2.0',
  'headers': [{'name': 'sec-ch-ua',
    'value': '"Not?A_Brand";v="8", "Chromium";v="108", "Google Chrome";v="108"'},
   {'name': 'Referer',
    'value': 'https://www.properly.ca/buy/search/toronto/?bounds=%7B%22sw%22%3A%7B%22lat%22%3A43.57821514343695%2C%22lng%22%3A-79.71032656047284%7D%2C%22ne%22%3A%7B%22lat%22%3A43.792721126438295%2C%22lng%22%3A-79.05114687297284%7D%7D&sorting=p

-Define df_text que vai ter os valores contidos na coluna 'text' do dataframe
-para cada linha 'i' no comprimento do dataframe df2: <br>
    -define js com os valores do df2 coluna text/linha i e retorna os valores contidos em 'response', 'content', e 'text' <br>
    -define json_object como um json serialiazed de js <br>
    -insere json_object na coluna text <br>

In [23]:
df_text = pd.DataFrame(columns = {'text':''})
for i in range(0,len(df2)):
    js = df2['text'][i].get('response',{}).get('content',{}).get('text')
    json_object = json.dumps(js)
    #item = str(i)
    df_text = df_text.append({'text':json_object}, ignore_index = True)

C:\Users\willb\AppData\Local\Temp\ipykernel_2656\3589460672.py:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df_text = df_text.append({'text':json_object}, ignore_index = True)
C:\Users\willb\AppData\Local\Temp\ipykernel_2656\3589460672.py:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df_text = df_text.append({'text':json_object}, ignore_index = True)
C:\Users\willb\AppData\Local\Temp\ipykernel_2656\3589460672.py:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df_text = df_text.append({'text':json_object}, ignore_index = True)
C:\Users\willb\AppData\Local\Temp\ipykernel_2656\3589460672.py:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat ins

In [24]:
js

''

In [25]:
df_text

text
0     "<!DOCTYPE html><html lang=\"en\"><head><meta ...
1     "!function(){\"use strict\";var e={},t={};func...
2     "\"use strict\";(self.webpackChunk_N_E=self.we...
3     "(self.webpackChunk_N_E=self.webpackChunk_N_E|...
4     "(self.webpackChunk_N_E=self.webpackChunk_N_E|...
...                                                 ...
4607                                                 ""
4608                                                 ""
4609                                                 ""
4610                                                 ""
4611                                                 ""

[4612 rows x 1 columns]

Esta contando quantas linhas identicas existem considerando os 20 primeiros caracteres de cada linha

In [26]:
df_text['text'].str[:20].value_counts()

null                    693
"{\"results\":[{\"fa    666
"R0lGODlhAQABAID/AP/    657
""                      631
"{\n  \"success\": t    353
"{\"vid\":\"779361aa    336
"success"               334
"{\"hits\":[{\"id\":    333
"AAABAAMAMDAAAAEAIAC    331
"{\"hits\":[{\"prope    116
"(self.webpackChunk_     31
"R0lGODlhAQABAIAAAAA     27
"\"use strict\";(sel     20
"1gca"                   10
"1d"                     10
"{\"pageProps\":{\"_      9
"OK"                      6
"\n// Copyright 2012      4
"(function(){var s =      3
"<!DOCTYPE html PUBL      3
"{\"ok\":true,\"isAc      2
"/**\n* Copyright (c      2
"/*! For license inf      2
"R0lGODlhAQABAIAAAP/      2
"!function(){\"use s      2
"(function(K){typeof      1
"(window.webpackJson      1
"{\"border\":\"#5F72      1
"!function(){var t,e      1
"{\"apiKey\": \"YmJk      1
"window['google-adwo      1
"{\"ok\":true,\"reco      1
"{\"pageProps\":{\"a      1
"{\"pageProps\":{\"p      1
"self.__SSG_MANIFEST      1
"self.__BUILD_MANIFE

Define df_hits, que eh df_text onde o string da coluna 'text' contem 'hits'

In [27]:
df_hits = df_text[df_text['text'].str.contains('hits')]
df_hits = df_hits.reset_index()
df_hits

index                                               text
0        4  "(self.webpackChunk_N_E=self.webpackChunk_N_E|...
1       28  "\"use strict\";(self.webpackChunk_N_E=self.we...
2       29  "(self.webpackChunk_N_E=self.webpackChunk_N_E|...
3      195  "{\"hits\":[{\"id\":\"mls-toronto-W5808402\",\...
4      243  "{\"hits\":[{\"id\":\"mls-toronto-X5772631\",\...
..     ...                                                ...
447   4544  "{\"hits\":[{\"id\":\"mls-toronto-S5684760\",\...
448   4557  "{\"hits\":[{\"id\":\"mls-toronto-N5790771\",\...
449   4570  "{\"hits\":[{\"id\":\"mls-toronto-E5801355\",\...
450   4583  "{\"hits\":[{\"id\":\"mls-toronto-C5794697\",\...
451   4589  "{\"hits\":[{\"property_location_id\":\"_uLcdU...

[452 rows x 2 columns]

df_hits em formato texto

In [28]:
df_hits.text

0      "(self.webpackChunk_N_E=self.webpackChunk_N_E|...
1      "\"use strict\";(self.webpackChunk_N_E=self.we...
2      "(self.webpackChunk_N_E=self.webpackChunk_N_E|...
3      "{\"hits\":[{\"id\":\"mls-toronto-W5808402\",\...
4      "{\"hits\":[{\"id\":\"mls-toronto-X5772631\",\...
                             ...                        
447    "{\"hits\":[{\"id\":\"mls-toronto-S5684760\",\...
448    "{\"hits\":[{\"id\":\"mls-toronto-N5790771\",\...
449    "{\"hits\":[{\"id\":\"mls-toronto-E5801355\",\...
450    "{\"hits\":[{\"id\":\"mls-toronto-C5794697\",\...
451    "{\"hits\":[{\"property_location_id\":\"_uLcdU...
Name: text, Length: 452, dtype: object

Os dados que queremos comecam depois da linha 29, entao definimos df_hits novamente como o proprio df-hits mas so com linhas em que o valor na coluna index seja maior que 29.

In [29]:
df_hits = df_hits[df_hits["index"] >29]
df_hits = df_hits.reset_index()
df_hits

level_0  index                                               text
0          3    195  "{\"hits\":[{\"id\":\"mls-toronto-W5808402\",\...
1          4    243  "{\"hits\":[{\"id\":\"mls-toronto-X5772631\",\...
2          5    249  "{\"hits\":[{\"property_location_id\":\"mXnbgb...
3          6    277  "{\"hits\":[{\"id\":\"mls-toronto-N5848462\",\...
4          7    283  "{\"hits\":[{\"property_location_id\":\"mXnbgb...
..       ...    ...                                                ...
444      447   4544  "{\"hits\":[{\"id\":\"mls-toronto-S5684760\",\...
445      448   4557  "{\"hits\":[{\"id\":\"mls-toronto-N5790771\",\...
446      449   4570  "{\"hits\":[{\"id\":\"mls-toronto-E5801355\",\...
447      450   4583  "{\"hits\":[{\"id\":\"mls-toronto-C5794697\",\...
448      451   4589  "{\"hits\":[{\"property_location_id\":\"_uLcdU...

[449 rows x 3 columns]

Quebrei o codigo abaixo pra testar

In [32]:
df_houses = pd.DataFrame(columns = {'id':'',
                                   'last_updated_str':'',
                                   'properly_city_code':'',
                                   'community':'',
                                   'neighbourhood_name':'',
                                   'neighbourhoods':'',
                                   'city':'',
                                   'province':'',
                                    '_geoloc':'',
                                    'formatted_address':'',
                                    'list_price':'',
                                    'on_market_date_str':'',
                                    'close_date_str':'',
                                    'property_sub_type':'',
                                    'levels':'',
                                    'parking_total':'',
                                    'living_area_sqft':'',
                                    'bedrooms_standard':'',
                                    'bedrooms_plus':'',
                                    'bedrooms_standard_and_plus':'',
                                    'bathrooms_full':'',
                                    'bathrooms_partial':'',
                                    'bathrooms_total':'',
                                    'images_count':'',
                                    'garage_spaces':'',
                                    'predicted_price':'',
                                    'error_rate':''})
df_houses

Empty DataFrame
Columns: [id, last_updated_str, properly_city_code, community, neighbourhood_name, neighbourhoods, city, province, _geoloc, formatted_address, list_price, on_market_date_str, close_date_str, property_sub_type, levels, parking_total, living_area_sqft, bedrooms_standard, bedrooms_plus, bedrooms_standard_and_plus, bathrooms_full, bathrooms_partial, bathrooms_total, images_count, garage_spaces, predicted_price, error_rate]
Index: []

[0 rows x 27 columns]

-cria um dataframe df_houses, que vai ter as colunas com os titulos listados, mas ainda sem valores (linhas) <br>
-deu pau no codigo ali, mas criou 'teste'que tera o valores de df_hits e deu uma limpada pra tirar '/' e afins <br>
-depois esta usando .append pra puxar os valores de teste para o df_houses criado acima, linha a linha





In [30]:
df_houses = pd.DataFrame(columns = {'id':'',
                                   'last_updated_str':'',
                                   'properly_city_code':'',
                                   'community':'',
                                   'neighbourhood_name':'',
                                   'neighbourhoods':'',
                                   'city':'',
                                   'province':'',
                                    '_geoloc':'',
                                    'formatted_address':'',
                                    'list_price':'',
                                    'on_market_date_str':'',
                                    'close_date_str':'',
                                    'property_sub_type':'',
                                    'levels':'',
                                    'parking_total':'',
                                    'living_area_sqft':'',
                                    'bedrooms_standard':'',
                                    'bedrooms_plus':'',
                                    'bedrooms_standard_and_plus':'',
                                    'bathrooms_full':'',
                                    'bathrooms_partial':'',
                                    'bathrooms_total':'',
                                    'images_count':'',
                                    'garage_spaces':'',
                                    'predicted_price':'',
                                    'error_rate':''})


                                    
for i in range(0,len(df_hits)):
    teste = df_hits['text'][i].replace('\\','').replace('\'','').replace('"{"hits":[','')
    teste = teste.split(',"nbHits"')
    teste = teste[0]
    teste = teste.replace('},{"id":"mls-','}|{"id":"mls-')
    teste = teste.split('|')
    for j in range(0,len(teste)-1):
        teste_2 = json.loads(teste[j])
        df_houses = df_houses.append({'id':teste_2.get('id'),
                                   'last_updated_str':teste_2.get('last_updated_str'),
                                   'properly_city_code':teste_2.get('properly_city_code'),
                                   'community':teste_2.get('community'),
                                   'neighbourhood_name':teste_2.get('neighbourhood_name'),
                                   'neighbourhoods':teste_2.get('neighbourhoods'),
                                   'city':teste_2.get('city'),
                                   'province':teste_2.get('province'),
                                    '_geoloc':teste_2.get('_geoloc'),
                                    'formatted_address':teste_2.get('formatted_address'),
                                    'list_price':teste_2.get('list_price'),
                                    'on_market_date_str':teste_2.get('on_market_date_str'),
                                    'close_date_str':teste_2.get('close_date_str'),
                                    'property_sub_type':teste_2.get('property_sub_type'),
                                    'levels':teste_2.get('levels'),
                                    'parking_total':teste_2.get('parking_total'),
                                    'living_area_sqft':teste_2.get('living_area_sqft'),
                                    'bedrooms_standard':teste_2.get('bedrooms_standard'),
                                    'bedrooms_plus':teste_2.get('bedrooms_plus'),
                                    'bedrooms_standard_and_plus':teste_2.get('bedrooms_standard_and_plus'),
                                    'bathrooms_full':teste_2.get('bathrooms_full'),
                                    'bathrooms_partial':teste_2.get('bathrooms_partial'),
                                    'bathrooms_total':teste_2.get('bathrooms_total'),
                                    'images_count':teste_2.get('images_count'),
                                    'garage_spaces':teste_2.get('garage_spaces'),
                                    'predicted_price':teste_2.get('predicted_price'),
                                    'error_rate':teste_2.get('error_rate')
                                   }, ignore_index = True)        

C:\Users\willb\AppData\Local\Temp\ipykernel_2656\696920814.py:36: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df_houses = df_houses.append({'id':teste_2.get('id'),
C:\Users\willb\AppData\Local\Temp\ipykernel_2656\696920814.py:36: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df_houses = df_houses.append({'id':teste_2.get('id'),
C:\Users\willb\AppData\Local\Temp\ipykernel_2656\696920814.py:36: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df_houses = df_houses.append({'id':teste_2.get('id'),
C:\Users\willb\AppData\Local\Temp\ipykernel_2656\696920814.py:36: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df_houses = df_houses.append({'id'

JSONDecodeError: Unterminated string starting at: line 1 column 2322 (char 2321)

In [33]:
df_houses

Empty DataFrame
Columns: [id, last_updated_str, properly_city_code, community, neighbourhood_name, neighbourhoods, city, province, _geoloc, formatted_address, list_price, on_market_date_str, close_date_str, property_sub_type, levels, parking_total, living_area_sqft, bedrooms_standard, bedrooms_plus, bedrooms_standard_and_plus, bathrooms_full, bathrooms_partial, bathrooms_total, images_count, garage_spaces, predicted_price, error_rate]
Index: []

[0 rows x 27 columns]

In [34]:
df_houses.to_parquet('df_houses.parquet')

ImportError: Unable to find a usable engine; tried using: 'pyarrow', 'fastparquet'.
A suitable version of pyarrow or fastparquet is required for parquet support.
Trying to import the above resulted in these errors:
 - Missing optional dependency 'pyarrow'. pyarrow is required for parquet support. Use pip or conda to install pyarrow.
 - Missing optional dependency 'fastparquet'. fastparquet is required for parquet support. Use pip or conda to install fastparquet.

In [35]:
teste_2

{'id': 'mls-toronto-X5809927',
 'property_location_id': 'qVHPhNfhSl6w982rPlboHA',
 'mls_id': 'X5809927',
 'standard_status': 'Closed',
 'last_updated': 1668209684,
 'last_updated_str': '2022-11-11 23:34:44+00',
 'google_place_id': 'ChIJg0AYme2IK4gRqk5fxtLBfwA',
 'building_place_id': 'ChIJg0AYme2IK4gRqk5fxtLBfwA',
 'properly_city_code': 'toronto-on-ca',
 'community': None,
 'neighbourhood_name': 'Hespeler Village',
 'neighbourhoods': [],
 'city': 'Cambridge',
 'province': 'ON',
 '_geoloc': {'lat': 43.427825, 'lng': -80.302224},
 'h3_geohash_r5': '852ab487fffffff',
 'h3_geohash_r6': '862ab4847ffffff',
 'h3_geohash_r7': '872ab4845ffffff',
 'h3_geohash_r8': '882ab48459fffff',
 'h3_geohash_r9': '892ab48458bffff',
 'association_fee': None,
 'association_fee_frequency': '',
 'formatted_address': '14 Alpine St, Cambridge, ON N3C 2J7, Canada',
 'list_price': 899900.0,
 'previous_list_price': None,
 'on_market_date': 1666915200,
 'on_market_date_str': '2022-10-28',
 'close_date': 1668124800,
 'c

In [ ]:
df_houses

In [36]:
teste = df_hits['text'][3].replace('\\','').replace('\'','').replace('"{"hits":[','')

In [37]:
teste

'{"id":"mls-toronto-N5848462","property_location_id":"F_SIxCO5SYKrhNdk4CkRUg","mls_id":"N5848462","standard_status":"Closed","last_updated":1672840736,"last_updated_str":"2023-01-04 13:58:56+00","google_place_id":"Eis2IEdyZWVuIEJyaWFyIFJkLCBOZXcgVGVjdW1zZXRoLCBPTiwgQ2FuYWRhIhoSGAoUChIJj_jlS2XpKogR9nG5mKEWRfgQBg","building_place_id":"Eis2IEdyZWVuIEJyaWFyIFJkLCBOZXcgVGVjdW1zZXRoLCBPTiwgQ2FuYWRhIhoSGAoUChIJj_jlS2XpKogR9nG5mKEWRfgQBg","properly_city_code":"toronto-on-ca","community":"Rural New Tecumseth","neighbourhood_name":"Rural New Tecumseth","neighbourhoods":[],"city":"New Tecumseth","province":"ON","_geoloc":{"lat":44.161005,"lng":-79.796894},"h3_geohash_r5":"852b9aabfffffff","h3_geohash_r6":"862b9aa17ffffff","h3_geohash_r7":"872b9aa12ffffff","h3_geohash_r8":"882b9aa12dfffff","h3_geohash_r9":"892b9aa12cfffff","association_fee":"595","association_fee_frequency":"Monthly","formatted_address":"6 Green Briar Rd, New Tecumseth, ON L0G, Canada","list_price":749000.0,"previous_list_price":n

In [ ]:
teste_list = teste.split(',"nbHits"')

In [ ]:
len(teste_list)

In [ ]:
teste_list2 = teste_list[0]

In [ ]:
teste_list3 = teste_list2.replace('},{"id":"mls-','}|{"id":"mls-')
teste_list3 = teste_list3.split('|')

In [ ]:
teste_list4 = json.loads(teste_list3[0])

In [ ]:
teste_list4.get('id')

In [ ]:
#teste_list4

In [ ]:
df_houses = pd.DataFrame(columns = {'id':'',
                                   'last_updated_str':'',
                                   'properly_city_code':'',
                                   'community':'',
                                   'neighbourhood_name':'',
                                   'neighbourhoods':'',
                                   'city':'',
                                   'province':'',
                                    '_geoloc':'',
                                    'formatted_address':'',
                                    'list_price':'',
                                    'on_market_date_str':'',
                                    'close_date_str':'',
                                    'property_sub_type':'',
                                    'levels':'',
                                    'parking_total':'',
                                    'living_area_sqft':'',
                                    'bedrooms_standard':'',
                                    'bedrooms_plus':'',
                                    'bedrooms_standard_and_plus':'',
                                    'bathrooms_full':'',
                                    'bathrooms_partial':'',
                                    'bathrooms_total':'',
                                    'images_count':'',
                                    'garage_spaces':'',
                                    'predicted_price':'',
                                    'error_rate':''})

In [ ]:
df_houses

In [ ]:
df_houses2 = df_houses.append({'id':teste_list4.get('id'),
                                   'last_updated_str':teste_list4.get('last_updated_str'),
                                   'properly_city_code':teste_list4.get('properly_city_code'),
                                   'community':teste_list4.get('community'),
                                   'neighbourhood_name':teste_list4.get('neighbourhood_name'),
                                   'neighbourhoods':teste_list4.get('neighbourhoods'),
                                   'city':teste_list4.get('city'),
                                   'province':teste_list4.get('province'),
                                    '_geoloc':teste_list4.get('_geoloc'),
                                    'formatted_address':teste_list4.get('formatted_address'),
                                    'list_price':teste_list4.get('list_price'),
                                    'on_market_date_str':teste_list4.get('on_market_date_str'),
                                    'close_date_str':teste_list4.get('close_date_str'),
                                    'property_sub_type':teste_list4.get('property_sub_type'),
                                    'levels':teste_list4.get('levels'),
                                    'parking_total':teste_list4.get('parking_total'),
                                    'living_area_sqft':teste_list4.get('living_area_sqft'),
                                    'bedrooms_standard':teste_list4.get('bedrooms_standard'),
                                    'bedrooms_plus':teste_list4.get('bedrooms_plus'),
                                    'bedrooms_standard_and_plus':teste_list4.get('bedrooms_standard_and_plus'),
                                    'bathrooms_full':teste_list4.get('bathrooms_full'),
                                    'bathrooms_partial':teste_list4.get('bathrooms_partial'),
                                    'bathrooms_total':teste_list4.get('bathrooms_total'),
                                    'images_count':teste_list4.get('images_count'),
                                    'garage_spaces':teste_list4.get('garage_spaces'),
                                    'predicted_price':teste_list4.get('predicted_price'),
                                    'error_rate':teste_list4.get('error_rate')
                                   }, ignore_index = True)

In [ ]:
df_houses2